lib/node_modules/@stdlib/math/stats/anova1/test/test.js

```diff
@@ -0,0 +1,184 @@
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			// Make the factor the same
+			anova1( value, ['frank', 'philipp'] );
+		};
+	}
+});
+
+tape( 'the function throws an error if the `x` argument has a length smaller than 2', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		[],
+		[5]
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			anova1( value, ['frank', 'philipp'] );
+		};
+	}
+});
+
+tape( 'the function throws an error if the `factor` argument is not an array', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		5,
+		true,
+		NaN,
+		null
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			anova1( [1, 2, 3, 4, 5], value );
+		};
+	}
+});
+
+tape( 'the function throws an error if the `factor` argument has a length smaller than 2', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		[],
+		[14],
+		['frank']
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			anova1( [1, 2, 3], value );
+		};
+	}
+});
+
+tape( 'the function throws an error if the `factor` argument has less than 2 unique elements', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		['frank', 'frank', 'frank'],
+		['alex', 'alex', 'alex', 'alex']
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), Error, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			anova1( [1, 2, 3], value );
+		};
+	}
+});
+
+tape( 'the function throws an error if the arguments have unequal lengths', function test( t ) {
+	// Two unique arguments for x and factor
+	var valuesX;
+	var valuesY;
+	var i;
+
+	valuesX = [
+		[1, 2, 3, 4, 5, 6, 7, 8],
+		[5, 6, 7, 4],
+		[1, 2, 3]
+	];
+
+	valuesY = [
+		['frank', 'alex', 'frank', 'alex'],
+		['philipp', 'alex', 'nugent'],
+		['john', 'delta beta']
+	];
+
+	for ( i = 0; i < valuesX.length; i++ ) {
+		t.throws( badValue( valuesX[i], valuesY[i] ), Error, 'throws an error when provided '+valuesX[i]+valuesY[i] );
+	}
+	t.end();
+
+	function badValue( valueX, valueY ) {
+		return function badValue() {
+			anova1( valueX, valueY );
+		};
+	}
+});
+
+tape( 'the function produces expected values on simple inputs', function test( t ) {
+	var out;
+	var x;
+	var f;
+
+	x = [3, 4, 5, 6, 2, 5, 10, 12, 8, 10];
+	f = ['frank', 'philipp', 'nugent', 'frank', 'philipp', 'nugent', 'frank', 'philipp', 'nugent', 'frank'];
+	out = anova1(x, f);
+
+	t.equal(out.treatmentDf, 2);
```

Comment:
Where do these values come from? Did you test against R? If so, please add a comment that this is the case such as `	// Tested against R:`. Even better would be to write an R script (based on the [R snippet](https://github.com/stdlib-js/stdlib/blob/develop/tools/snippets/test/fixtures/r/runner.R)) to automatically generate these test fixtures. The script would save them as JSON, which we would then load in the tests file to compare against. This is preferable as it makes the tests entirely reproducible. It is more work, admittedly, which is why I haven't always done it so far either. Doesn't mean we shouldn't, though.

Reply:
Comment added. @TODO Will add an R fixture at a later date.

Reply:
Just an FYI: be careful when using the `@` symbol on GitHub. It should only be used when explicitly referring to a GitHub user. And I am pretty sure the user [`@todo`](https://github.com/todo) was **not** supposed to be tagged in this PR.